src/main/java/org/apache/commons/collections4/bloomfilter/LayerManager.java

```diff
@@ -179,15 +190,15 @@ public static final class ExtendCheck {
          * @return A Predicate suitable for the LayerManager {@code extendCheck} parameter.
          * @throws IllegalArgumentException if {@code breakAt <= 0}
          */
-        public static Predicate<LayerManager> advanceOnCount(int breakAt) {
+        public static <T extends BloomFilter> Predicate<LayerManager<T>> advanceOnCount(int breakAt) {
             if (breakAt <= 0) {
                 throw new IllegalArgumentException("'breakAt' must be greater than 0");
             }
-            return new Predicate<LayerManager>() {
+            return new Predicate<LayerManager<T>>() {
                 int count;
 
                 @Override
-                public boolean test(LayerManager filter) {
+                public boolean test(LayerManager<T> filter) {
                     return ++count % breakAt == 0;
```

Comment:
Just noticed this use of modulus. It will be incorrect when the counter overflows. In the following example using mod 3 an extra call is required to reach modulus zero; with 7 it takes 3 calls too few:
```
jshell> int i = Integer.MAX_VALUE
i ==> 2147483647

jshell> i % 3
$2 ==> 1       // Should be 2 calls from here

jshell> (i+1) % 3
$3 ==> -2

jshell> (i+2) % 3
$4 ==> -1

jshell> (i+3) % 3
$5 ==> 0


jshell> i % 7
$10 ==> 1       // Should be 6 calls from here

jshell> (i+1) % 7
$11 ==> -2

jshell> (i+2) % 7
$12 ==> -1

jshell> (i+3) % 7
$13 ==> 0
```

It can be fixed using:
```java
public boolean test(LayerManager<T> filter) {
    if (++count == breakAt) {
        count = 0;
        return true;
    }
    return false;
}
```

The use of a branch statement _may_ be offset by avoiding a modulus division.

Q. Is it realistic to call the extend check 2 billion times? Will anyone notice if the extend check is incorrect after 2 billion calls?

Still, I would prefer to not include a known bug in the release however trivial the consequences of it may be.